src/mgr/PyModuleRegistry.cc

```diff
@@ -410,6 +410,29 @@ void PyModuleRegistry::upgrade_config(
       auto last_slash = i.first.rfind('/');
       const std::string module_name = i.first.substr(4, i.first.substr(4).find('/'));
       const std::string key = i.first.substr(last_slash + 1);
+
+      const auto &value = i.second;
+
+      // Heuristic to skip things that look more like stores
+      // than configs.
+      bool is_config = true;
+      for (const auto &c : value) {
+        if (c == '\n' || c == '\r' || c < 0x20) {
+          is_config = false;
+          break;
+        }
+      }
+
+      if (value.size() > 256) {
+        is_config = false;
+      }
+
+      if (!is_config) {
+        dout(1) << "Not migrating config module:key "
+                << module_name << " : " << key << dendl;
+        continue;
+      }
+
```

Comment:
This doesn't remove the old "mgr/*" keys that get moved, right?  It probably should, though?

It might also be worth noting in a comment that this is non-atomic.. if it gets interrupted the first time it runs it won't ever finish the migration/conversion.  Maybe it's better to have an explicit marker somewhere that the conversion has completed... though I'm not sure where to do that in a way that doesn't clutter things up going forward.  Maybe a mgr_config_style=mimic key is worthwhile as it'll let us change things again later if we need to.

Reply:
The old keys don't get removed because the conversion is never quite sure if it's looking at real configuration or at things that should be left for get_store/set_store.  I'm inclined to add in a very basic config schema to the modules (perhaps just the list of keys) so that they can disambiguate this, if we can sneak it into mimic.

Reply:
That sounds good to me.  This would get recorded... in the MgrMap, along with the avaialble/active modules there?  It would be nice to get the config help and validation too (i.e., full schema).  Can push that off if it's too involved, though; just need to make sure the encoding versions easily.